src/merkle/merkle-objects.ts

```diff
@@ -202,10 +202,11 @@ export class IpfsMerge implements MergeFunction<CIDHolder, TreeMetadata> {
 
   async merge(
     left: Node<CIDHolder>,
-    right: Node<CIDHolder>,
+    right: Node<CIDHolder> | null,
     metadata: TreeMetadata | null
   ): Promise<Node<CIDHolder>> {
-    const merged = [left.data.cid, right.data.cid]
+    const merged = [left.data.cid, right?.data?.cid]
```

Comment:
```suggestion
    const merged = [left.data.cid, right?.data?.cid || null]
```
Sorry, the version I suggested would wind up putting `undefined` into the tree instead of null, which I don't think can even be expressed in IPLD